### PR TITLE
Support external redis host and using 127.0.0.1 as default

### PR DESCRIPTION
**Pull request by p4xx07**
I want to use my external redis because after many hours of streaming and recording oryx stops working. There are many keys written that get cleaned only after you delete the recording task.
I would like to be able to use my custom redis to support more intensive recordings

### CODE REVIEW

File: platform/utils.go [hunk 1]
```diff
@@ -481,7 +485,11 @@ var rdb *redis.Client
 
 // InitRdb create and init global rdb, which is a redis client.
 func InitRdb() error {
-	addr := "127.0.0.1"
+	addr := envRedisAddress()
+	if addr == "" {
+		addr = "127.0.0.1"
```
Comment (winlinvip) [inline platform/utils.go:490]: You should set the default env to "127.0.0.1", for example:

```go
setEnvDefault("REDIS_HOST", "127.0.0.1")
```

See also:

```go
setEnvDefault("REDIS_PORT", "6379")
```

File: platform/utils.go [hunk 2]
```diff
@@ -436,6 +436,10 @@ func envRedisPort() string {
 	return os.Getenv("REDIS_PORT")
 }
 
+func envRedisAddress() string {
+	return os.Getenv("REDIS_ADDRESS")
```
Comment (winlinvip) [inline platform/utils.go:440]: Please rename to `REDIS_HOST` and `envRedisHost()` instead.

**Comment by winlinvip** [comment]
我想这个需求是合理的，感谢你提交这个PullRequest。

同时，我想和你讨论一个问题：如果使用外部的Redis Server，我们是否应该支持databases，这样可以多个Oryx使用同样的一个redis，通过参数DB隔离数据：

```go
redis.NewClient(&redis.Options{Addr, Password, DB: 0})
```

如果你觉得合理，那么请增加一个环境变量 `REDIS_DATABASE`，默认为0。

**Comment by p4xx07** [comment]
Ok I think we can do this. Let me make the changes

**Comment by p4xx07** [comment]
I've also attempted to run this docker with my modifications, but I noticed even if I have my own redis setup, oryx will attempt to create it's own redis server. How should we face this issue? Do you have any ideas?

**Comment by Allan-Nava** [comment]
> I've also attempted to run this docker with my modifications, but I noticed even if I have my own redis setup, oryx will attempt to create it's own redis server. How should we face this issue? Do you have any ideas?

Same problem

**Comment by winlinvip** [comment]
> @winlinvip let me know if we need to do more changes.
> 
> The issue I have been having is that if I record multiple streams for more than 10 hours with a docker with 4GB of ram, Oryx will start writing ts files in the wrong positions in the filesystem, so I'm guessing that redis gets filled up and oryx can't keep track of what he is supposed to do. Oryx will also stop working and won't come back up. If you have any thoughts on how to solve this it would be great. I'm hoping that outsourcing redis from the docker will fix this issue.

这个问题和这个PR不相关，请单独提交一个Issue详细说明它。

**Comment by winlinvip** [comment]
> I've also attempted to run this docker with my modifications, but I noticed even if I have my own redis setup, oryx will attempt to create it's own redis server. How should we face this issue? Do you have any ideas?

这个环境变量只是决定了Oryx连接哪个Redis server，但Oryx永远会创建一个自己的redis server，这个应该不影响最终的效果，它占用的资源也非常少，只是不使用而已。

File: platform/utils.go [hunk 3]
```diff
@@ -481,11 +489,15 @@ var rdb *redis.Client
 
 // InitRdb create and init global rdb, which is a redis client.
 func InitRdb() error {
-	addr := "127.0.0.1"
+	redisDatabase, err := strconv.Atoi(envRedisDatabase())
+	if err != nil {
+		return errors.Wrapf(err, "invalid REDIS_DATABASE")
```
Comment (winlinvip) [inline platform/utils.go:494]: 错误对象中，应该包含完善的错误信息：

```go
return errors.Wrapf(err, "invalid REDIS_DATABASE %v", envRedisDatabase())
```

File: platform/utils.go [hunk 4]
```diff
@@ -436,6 +436,14 @@ func envRedisPort() string {
 	return os.Getenv("REDIS_PORT")
 }
 
+func envRedisHost() string {
```
Comment (winlinvip) [inline platform/utils.go:439]: 应该打印出这个环境变量的值，参考 `envRedisPort` 的用法，这个是重要的日志，有可能用户会设置错误。

PS：同样也要打印出 `envRedisDatabase` 的值。

Comment (p4xx07) [reply platform/utils.go:439]: Ok I included your suggestions

**Comment by p4xx07** [comment]
> > @winlinvip let me know if we need to do more changes.
> > The issue I have been having is that if I record multiple streams for more than 10 hours with a docker with 4GB of ram, Oryx will start writing ts files in the wrong positions in the filesystem, so I'm guessing that redis gets filled up and oryx can't keep track of what he is supposed to do. Oryx will also stop working and won't come back up. If you have any thoughts on how to solve this it would be great. I'm hoping that outsourcing redis from the docker will fix this issue.
> 
> This issue is not related to this pull request. Please submit a separate issue with detailed information about it.
> 
> `TRANS_BY_GPT4`

I have created a seperate issue with screenshots detailing the issue. Thanks @winlinvip 

https://github.com/ossrs/oryx/issues/197

**Pull request merged**